corehq/apps/es/es_query.py

```diff
@@ -466,6 +466,12 @@ def values_list(self, *fields, **kwargs):
     def count(self):
         return self.adapter.count(self.raw_query)
 
+    def exists(self):
+        """Checks to see whether any documents match the query"""
+        query = self.size(0)
+        query.es_query['terminate_after'] = 1
```

Comment:
nit: can the same improvement be made on function, get_last_form_submission_received?
This would be a great benefit where we get all matches to just pick the first one or say first 10/20/50/100 in case of searching like in CaseList?